Compactable/Type

```diff
@@ -15,16 +15,5 @@ Since both functions can be written in terms of the other, we can create instanc
 of the function fromCompact or fromSeparate.
 
 -}
-    let Either = ./../Either/Type
-
-in    λ(f : Type → Type)
-    → { mapOptional :
-          ∀(a : Type) → ∀(b : Type) → (a → Optional b) → f a → f b
-      , mapEither :
```

Comment:
I was wrong, you did derive `mapEither`! Forget I said anything.

Reply:
We can derive `mapEither` from `mapOptional` it's [here](https://github.com/FormationAI/dhall-bhat/blob/92058454998868f9a3a67cd44bf50dbd8534f495/Compactable/mapEither) :) The motivation was that `mapOptional` is the canonical function when thinking about this class in terms of categories: `Compactable f === KleisliFunctor Optional f` where `kmap :: (a -> m b) -> f a -> f b` is specialised to `(a -> Optional b) -> f a -> f b` i.e. `mapOptional` :tada:! 

This will make future integration with `catewaul` easier :)

Reply:
Heh! I wrote my response before I saw the next comment :smile: